src/ros_filter.cpp

```diff
@@ -1916,6 +1916,15 @@ namespace RobotLocalization
       {
         ROS_ERROR_STREAM("Critical Error, NaNs were detected in the output state of the filter." <<
               " This was likely due to poorly conditioned process, noise, or sensor covariances.");
+        if(last_odom.header.stamp.sec > 0.0)
+        {
+          ROS_ERROR_STREAM("Apply Previous Pose");
```

Comment:
Can we verify that this case gets triggered by the test?

Reply:
https://github.com/burro-robotics/burro/pull/4229#:~:text=with%20fix-,%24%20rostest%20burro_tests%20ukf_nan_handling.test%20%2D%2Dreuse%2Dmaster,-...%20logging%20to%20/home